sklearn/ensemble/tests/test_iforest.py

```diff
@@ -134,14 +127,12 @@ def test_iforest_error():
     # test X_test n_features match X_train one:
     assert_raises(ValueError, IsolationForest().fit(X).predict, X[:, 1:])
 
-    # test threshold_ attribute error when behaviour is not old:
-    msg = "threshold_ attribute does not exist when behaviour != 'old'"
-    assert_raises_regex(AttributeError, msg, getattr,
-                        IsolationForest(behaviour='new'), 'threshold_')
+    # test that behaviour='old' will raise an error
```

Comment:
maybe add `# XXX to be removed in 0.24`

Reply:
actually the test will fail when behaviour will be removed in 0.24 so we won't miss to remove this as well.

Reply:
Yep this is a `NotImplementedError` so we will not miss it